### PR TITLE
Meta var for skip cache read for request

### PR DESCRIPTION
**Pull request by tonal**

### CODE REVIEW

**Comment by kmike** [comment]
@tonal could you please explain what is the use case for this, i.e. why do you want to write to the cache, but not to read from it?

**Comment by nramirezuy** [comment]
I think that the best way for this is for you extending the middleware on your side. 

```python
    def process_request(self, request, spider):
        # Skip uncacheable requests
        if request.meta.get('dont_cache_read'):
            self.stats.inc_value('httpcache/dont_read', spider=spider)
            request.meta['dont_cache'] = True
        result = super(CustomCacheMdw, self).process_request(request, spider)
        if request.meta.get('dont_cache_read'):
            request.meta.pop('dont_cache')
        return result
```

**Comment by tonal** [comment]
@kmike For example 1: if work with proxy list. Some proxy return status 502, 503, and other. It do`nt need save. Bat if return status 200 or 404 - save it. In middleware analize responce and Retray request without read in cache.
Example 2: Some site without cookie return simple JS code. It set cookie and refresh page. In callback I parse this JS, get this cookie, and return new Request with it. Response need save.

**Comment by Digenis** [comment]
For both examples this feature would only serve as a workaround.

However see #1840 and my comment there.
I'd reword part of my comment as follows:
> This is a typical problem where deciding whether
* ~~to retry a request~~
* ~~to cache a response~~
* **a response was a failure**

> is unavoidably tied to spider/parsing logic.

It's very easy to pollute response.meta with reserved keys
if we go on implementing something like this for every middleware
for whatever can go wrong.

I have some ideas on that PR
but for now, your fastest solution would be @nramirezuy's suggestion.
